### PR TITLE
<Calendar/> - use <DropdownBase/> internally

### PR DESCRIPTION
**Pull request by EgoziE**
This PR is a bit complicated so I'll explain.
The main goal is removing `<WixComponent/>`. 🏆 
The only component using it left is `<DropdownLayout/>`, so in order for it to be removed, we need to remove `onClickOutside` functionality that is not relevant for `<DropdownLayout/>` because it's a layout component.
The only component using `onClickOutside` functionality from `<DropdownLayout/>` is the calendar.
So in this PR I did the following:
1. In Calendar - replacing `<Text/>` + `<DropdownLayout/>` (without any popover) with `<TextButton/>` + `<DropdownBase/>` (popover inside DropdownBase).
2. In addition to the first point, adjusted some visuals according to Zeplin.
3. Added a missing prop (focusOnSelectedOption) to DropdownBase (that is passed to Popover)
4. Added a deprecation log for `onClickOutside` in `<DropdownLayout/>`

### CODE REVIEW

**Comment by EgoziE** [comment]
The relevant dropdowns I replaced are the month and year:
image

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5829/

File: src/DropdownBase/DropdownBase.private.driver.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import dropdownLayoutDriverFactory from '../DropdownLayout/DropdownLayout.driver';
+
+const dropdownBaseDriverFactory = ({ element }) => {
+  return {
+    exists: () => !!element,
+    element: () => element,
+    getDropdownLayoutDriver: () => {
+      return dropdownLayoutDriverFactory({
+        element: element.querySelector(
```
Comment (shlomitc) [inline src/DropdownBase/DropdownBase.private.driver.js:9]: change to a `unidriver` syntax

File: src/DropdownLayout/DropdownLayout.js [hunk 2]
```diff
@@ -38,6 +38,12 @@ class DropdownLayout extends WixComponent {
       hovered: NOT_HOVERED_INDEX,
       selectedId: props.selectedId,
     };
+
+    if (props.hasOwnProperty('onClickOutside')) {
+      deprecationLog(
+        'onClickOutside prop is deprecated in DropdownLayout and will be removed soon, please use dropdown base instead.',
```
Comment (shlomitc) [inline src/DropdownLayout/DropdownLayout.js:44]: add the component name at the beginning to indicate it to the consumers

**Pull request merged**